src/install/mac-install.ts

```diff
@@ -16,32 +16,33 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 
-import * as path from 'node:path';
-import * as fs from 'node:fs';
 import * as os from 'node:os';
 
 import * as extensionApi from '@podman-desktop/api';
 import { compare } from 'compare-versions';
-import { BaseCheck, BaseInstaller } from './base-install';
-import { getAssetsFolder, runCliCommand } from '../util';
+import { BaseCheck, BaseInstaller, CrcReleaseInfo } from './base-install';
+import { getAssetsFolder, isFileExists, runCliCommand } from '../util';
 
+const macosInstallerFineName = 'crc-macos-installer.pkg';
 
 export class MacOsInstall extends BaseInstaller {
-  install(logger?: extensionApi.Logger): Promise<boolean> {
+
+  install(releaseInfo: CrcReleaseInfo, logger?: extensionApi.Logger): Promise<boolean> {
     return extensionApi.window.withProgress({ location: extensionApi.ProgressLocation.APP_ICON }, async progress => {
       progress.report({ increment: 5 });
+      
+      const pkgPath = await this.downloadAndCheckInstaller(releaseInfo.links.darwin, macosInstallerFineName);
 
-      const pkgPath = path.resolve(getAssetsFolder(), 'crc-macos-installer.pkg');
       try {
-        if (fs.existsSync(pkgPath)) {
+        if (await isFileExists(pkgPath)) {
```

Comment:
is this related to the main ui async issue you mentioned?

Reply:
yes, I made a lot of `sync` code clean up, ensuring that we don’t use any synchronous code/IO operation.